### PR TITLE
Source Slack: high test strictness level

### LINKED ISSUE #19064: Source Slack: enable `high` test strictness level in SAT
## What
A `test_strictness_level` field was introduced to Source Acceptance Tests (SAT).
Slack is a generally_available connector, we want it to have a `high` test strictness level.

**This will help**:
- maximize the SAT coverage on this connector.
- document its potential weaknesses in term of test coverage.

## How
1. Migrate the existing `acceptance-test-config.yml` file to the latest configuration format. (See instructions [here](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/bases/source-acceptance-test/README.md#L61))
2. Enable `high` test strictness level in `acceptance-test-config.yml`. (See instructions [here](https://github.com/airbytehq/airbyte/blob/master/docs/connector-development/testing-connectors/source-acceptance-tests-reference.md#L240))
3. Commit changes on `acceptance-test-config.yml` and open a PR.
4. Run SAT with the `/test` command on the branch.
5. If tests are failing please fix the failing test or use `bypass_reason` fields to explain why a specific test can't be run.

### PR DESCRIPTION
**Pull request by davydov-d**
## What
https://github.com/airbytehq/airbyte/issues/19064

## How
- Modify abnormal state
- Add configured catalogs
- Do not join channels that not gonna be synced
- Refactor acceptance test config

### CODE REVIEW

**Review by lazebnyi — APPROVED** [review]

File: airbyte-integrations/connectors/source-slack/acceptance-test-config.yml [hunk 1]
```diff
@@ -1,29 +1,34 @@
 connector_image: airbyte/source-slack:dev
-tests:
+test_strictness_level: high
```
Comment (evantahler) [inline airbyte-integrations/connectors/source-slack/acceptance-test-config.yml:2]: 👍 no bypass reason

File: airbyte-integrations/connectors/source-slack/integration_tests/expected_records.jsonl [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
... "image_24": "https://secure.gravatar.com/avatar/95f67810af139bb2658d257c02efed94.jpg?s=24&d=https%3A%2F%2Fa.slack-edge.com%2Fdf10d%2Fimg%2Favatars%2Fava_0006-24.png", "image_32": "https://secure.gravatar.com/avatar/95f67810af139bb2658d257c02efed94.jpg?s=32&d=https%3A%2F%2Fa.slack-edge.com%2Fdf10d%2Fimg%2Favatars%2Fava_0006-32.png", "image_48": "https://secure.gravatar.com/avatar/95f67810af139bb2658d257c02efed94.jpg?s=48&d=https%3A%2F%2Fa.slack-edge.com%2Fdf10d%2Fimg%2Favatars%2Fava_0006-48.png", "image_72": "https://secure.gravatar.com/avatar/95f67810af139bb2658d257c02efed94.jpg?s=72&d=https%3A%2F%2Fa.slack-edge.com%2Fdf10d%2Fimg%2Favatars%2Fava_0006-72.png", "image_192": "https://secure.gravatar.com/avatar/95f67810af139bb2658d257c02efed94.jpg?s=192&d=https%3A%2F%2Fa.slack-edge.com%2Fdf10d%2Fimg%2Favatars%2Fava_0006-192.png", "image_512": "https://secure.gravatar.com/avatar/95f67810af139bb2658d257c02efed94.jpg?s=512&d=https%3A%2F%2Fa.slack-edge.com%2Fdf10d%2Fimg%2Favatars%2Fava_0006-512.png", "status_text_canonical": "", "team": "T04KX3KDDU6"}, "is_bot": false, "is_app_user": false, "updated": 1675090829, "is_forgotten": true, "is_invited_user": true}, "emitted_at": 1675094014038}
```
Comment (evantahler) [inline airbyte-integrations/connectors/source-slack/integration_tests/expected_records.jsonl:9]: 👍 fake data

**Review by evantahler — APPROVED** [review]

**Pull request merged**